chain/consensus/hierarchical/actors/sca/sca_apply.go

```diff
@@ -41,6 +52,67 @@ func applyFund(rt runtime.Runtime, msg types.Message) {
 	}
 }
 
+func applyRelease(rt runtime.Runtime, msg types.Message) {
+	var st SCAState
+
+	rfrom, rto := fromToRawAddr(rt, msg.From, msg.To)
+	snFrom, err := msg.From.Subnet()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error getting subnet from HAddress")
+
+	// Check that msg.From == BurntRewardsActorAddr, this determines that is a release message
+	if rfrom != builtin.BurntFundsActorAddr {
+		rt.Abortf(exitcode.ErrIllegalArgument, "msg passed as argument not a funding message (from != burtnRewardActorAddr)")
+	}
+
+	rt.StateTransaction(&st, func() {
+		bottomUpStateTransition(rt, &st, msg)
+
+		// Update circulating supply reducing release value.
+		sh, has, err := st.GetSubnet(adt.AsStore(rt), snFrom)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "error fetching subnet state")
+		if !has {
+			rt.Abortf(exitcode.ErrIllegalState, "subnet for actor hasn't been registered yet")
+		}
+		// Even if the actor has balance, we shouldn't allow releasing more than
+		// the current circulating supply, as it would mean that we are
+		// releasing funds from the collateral.
+		if sh.CircSupply.LessThan(msg.Value) {
+			rt.Abortf(exitcode.ErrIllegalState, "wtf! we can't release funds below the circulating supply. Something went wrong!")
+		}
+		sh.CircSupply = big.Sub(sh.CircSupply, msg.Value)
+		st.flushSubnet(rt, sh)
```

Comment:
can the flush fail asynchronously? We might need to block and wait to make sure the substraction has taken effect before releasing the funds